### PR TITLE
Vv/#178 implement videos update cli

### LINKED ISSUE #178: 🎁 New feature: CLI for extra details
# My Problem

I need to automate the process of collecting details about videos on Youtube.

# Solution

Use the official YT API to go over all the videos that we have saved in the JSON files and pull additional detials:

- Title
- Description
- Time duration
- Date of uploading
- Tags

How this CLI should work: 

- The code loads all the JSON files in to memory
- Loops over the objects and moves to another array the object with missing information 
- Loops over those by making query to YT, and updates the objects (overwrites everything, even if the data exists)
- Saves back all the objects back to disk.

This means that when you where to run it now the first time, all the data will be overwritten and updated, this way we can remove any mistakes that I have done.

And in the future when you add just the YT URL, you can run the CLI later to pull all the extra details, since the rule is, use the object even when one thing is missing.

IMPORTANT: remove the thumbnail key, since it is not useful - I think

### PR DESCRIPTION
**Pull request by acidbiscuit**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] Documentation content changes
- [ ] Other (please describe):

## Description

This PR includes 2 major changes:
1) Database format change for stored videos.
Now videos include `service` and `videoId` fields.
-  `service` contains service slug in order to ensure that video has a relation to certain service regardless of folder structure. This will be useful for the next CLI to verify that video belongs to one or another service. 
- `videoId` field contains valid Youtube video ID. It comes as a replacement for `url` and `thumbnail` fields. This improvements allows to reduce amount of data stored (by removing URL) and allows to remove all the steps to extract video id from the URL, but use it directly for displaying embed, showing thumbnail and fetching details from Youtube API.

2) CLI script to search videos with missing details, request relevant video information and update local database for each service.
Check out documented CLI script [execution flow](https://github.com/aws-videos-catalog/avc_website/blob/076396f6d4eacd175d95f3d60883afbc5f877ede/cli/videos-update/README.md).  

Note that we now use Youtube API client to consume Youtube Data API v3 and get videos details. Youtube client requires API key from Google Developers Console. Check out updated project's [README](https://github.com/aws-videos-catalog/avc_website/blob/076396f6d4eacd175d95f3d60883afbc5f877ede/README.md) on how to provide credentials to access Youtube API. Additional configuration of environment variables is required to prevent exposure of sensitive data in public repository since Youtube API allows only 10000 requests per day for free and may charge for additional requests if account billing is enabled.

All the videos with missing details and update results are logged into console. Once script execution is finished, all the JSON files are updated in place for corresponding service.

## What is the new behavior?

- Format of database storage for videos is changed
- New `npm` command is available to run the cli `npm run cli:videos-update`

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

## Other information
Note: `npm install` is required

**Issue(s) this PR closes**: Closes #178

### CODE REVIEW

**Comment by davidgatti** [comment]
Wow fantastic it just worked! :) Can you explain in the README what this means?

```
Unable to locate video details 3xzhW-2aD4s. Skipping
Unable to locate video details PVeDVz-5SRI. Skipping
```

What dose it mean? the video is no longer available on Youtube, Google returned a error, and I should run the script one more time to get all the videos? Or something else.

**Comment by acidbiscuit** [comment]
@davidgatti Thanks :)

I will check possibility to display actual response error from Youtube. Will have to postpone until tomorrow though. I've exceeded my quotas for today and beyond).

Will also update README as requested.

**Comment by davidgatti** [comment]
Sounds good.

**Comment by acidbiscuit** [comment]
@davidgatti I've added better log message text and an explanation in documentation.  

In our case, this error means that video id is valid but Youtube does not provide video details, whether it was removed or set to private.

File: cli/videos-update/index.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+/**
+ * Fetch and format details for each provided video
+ * @param unfilledVideos - list of videos to fetch details for
+ * @param serviceSlug
+ * @returns {Promise<unknown>}
+ */
+async function fetchVideosDetails (unfilledVideos, serviceSlug) {
+  return async.mapLimit(unfilledVideos, 1, async (video, callback) => {
+    try {
+      const videoDetails = await YoutubeClient.getVideoDetails('3xzhW-2aD4s')
```
Comment (davidgatti) [inline cli/videos-update/index.js:35]:
```suggestion
      const videoDetails = await YoutubeClient.getVideoDetails(video.videoId)
```

If you apply this fix, the CLI returns: 

```
Unable to load video details. Video is private or doesn't exist. Skipping undefined
GaxiosError: No filter selected. Expected one of: id, myRating, chart
    at Gaxios._request (/Users/davidgatti/Documents/GitHub/AVC/avc_website/node_modules/gaxios/src/gaxios.ts:117:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
```

**Review by davidgatti — CHANGES_REQUESTED** [review]

**Comment by acidbiscuit** [comment]
@davidgatti 
I've made a mistake in the latest commit. Please, check updated version.  
There were also a few videos with old format after I've merged with master branch. Those were causing exceptions and now handled appropriately.

**Pull request merged**